gr-trellis/lib/interleaver.cc

```diff
@@ -113,13 +113,13 @@ namespace gr {
       std::vector<int> tmp(d_K);
       unsigned char *bytes = reinterpret_cast<unsigned char*>(&tmp[0]);
 
-      for(unsigned int i = 0; i < d_K; i += 8) {
+      for(int i = 0; i < d_K; i += 8) {
```

Comment:
This single change is surprising, as it takes unsignedness away from something that just counts upwards; I'll do the following: I'll merge your change, but will then consistify `d_K` to be unsigned all throughout the trellis interleaver usage.

Reply:
d_k is declared as int and there are a lot of comparisons to int's.
For instance https://github.com/gnuradio/gnuradio/blob/b00ebcae4cf2047ba9bf8d7d2d16f9053fcd45b7/gr-trellis/lib/interleaver.cc#L87

So it makes sence to remove the 'unsigned' at this point.

On the other side d_k may not become negative, so d_k could be changed to unsigned int.
If you think that will increase code quality, I could change this, but this will have effects even to calling functions.